src/util/ActiveShapeUtils.tsx

```diff
@@ -74,15 +82,17 @@ export function Shape<OptionType, ExtraProps, ShapePropsType>({
   ...props
 }: ShapeProps<OptionType, ExtraProps, ShapePropsType>) {
   let shape: React.JSX.Element;
+  // eslint-disable-next-line one-var
+  let nextProps: ShapePropsType;
 
-  if (isValidElement(option)) {
-    shape = cloneElement(option, props);
+  if (isValidElement<OptionType>(option)) {
+    nextProps = propTransformer(option.props, props);
+    shape = cloneElement<OptionType>(option, nextProps);
```

Comment:
even though we remove `as` this is still casting.

Its probably fine since we're doing all of the checks we do here but still

Reply:
Made another update check again, please.